// n8n-nodes-soaprequest/dist/nodes/SoapRequest/Authentication.operations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.authenticationOperations = exports.SOAP_API_CREDENTIALS_TYPE = exports.SOAP_API_CREDENTIAL = exports.AUTHENTICATION = void 0;
exports.AUTHENTICATION = 'authentication';
exports.SOAP_API_CREDENTIAL = 'soapApi';
exports.SOAP_API_CREDENTIALS_TYPE = 'soapAPICredentialType';
exports.authenticationOperations = [
    {
        displayName: 'Authentication',
        name: exports.AUTHENTICATION,
        noDataExpression: true,
        type: 'options',
        options: [
            {
                name: 'None',
                value: 'none',
            },
            {
                name: 'SOAP API Credential Type',
                value: exports.SOAP_API_CREDENTIALS_TYPE,
            }
        ],
        default: 'none',
    },
    {
        displayName: 'SOAP Api Credentials',
        name: exports.SOAP_API_CREDENTIAL,
        type: 'credentials',
        noDataExpression: true,
        required: true,
        default: '',
        displayOptions: {
            show: {
                authentication: [exports.SOAP_API_CREDENTIALS_TYPE],
            },
        },
    },
];
