// n8n-nodes-soaprequest/dist/nodes/SoapRequest/Soap.operations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.soapOperations = exports.URL = exports.SOAP_HEADERS = exports.SOAP_BODY = exports.SEND_HEADERS = exports.METHOD = exports.CONVERT_TO_JSON = void 0;
const requestMethods = [
    {
        name: 'DELETE',
        value: 'DELETE',
    },
    {
        name: 'GET',
        value: 'GET',
    },
    {
        name: 'HEAD',
        value: 'HEAD',
    },
    {
        name: 'OPTIONS',
        value: 'OPTIONS',
    },
    {
        name: 'PATCH',
        value: 'PATCH',
    },
    {
        name: 'POST',
        value: 'POST',
    },
    {
        name: 'PUT',
        value: 'PUT',
    },
];
exports.CONVERT_TO_JSON = 'convertToJson';
exports.METHOD = 'method';
exports.SEND_HEADERS = 'sendHeaders';
exports.SOAP_BODY = 'soapBody';
exports.SOAP_HEADERS = 'soapHeaders';
exports.URL = 'url';
exports.soapOperations = [
    {
        displayName: 'Method',
        description: 'The request method',
        name: exports.METHOD,
        default: 'GET',
        options: requestMethods,
        type: 'options',
    },
    {
        displayName: 'URL',
        description: 'The request URL',
        default: '',
        name: exports.URL,
        required: true,
        type: 'string'
    },
    {
        displayName: 'Send Headers',
        name: exports.SEND_HEADERS,
        type: 'boolean',
        default: false,
        noDataExpression: true,
        description: 'Whether the SOAP request has a header section or not',
    },
    {
        displayName: 'Headers',
        name: exports.SOAP_HEADERS,
        type: 'string',
        typeOptions: {
            rows: 3,
            alwaysOpenEditor: false,
            editor: 'jsEditor'
        },
        displayOptions: {
            show: {
                sendHeaders: [true]
            },
        },
        default: '',
    },
    {
        displayName: 'Body',
        default: '',
        name: exports.SOAP_BODY,
        required: true,
        type: 'string',
        typeOptions: {
            rows: 3,
            alwaysOpenEditor: false,
            editor: 'jsEditor'
        }
    },
    {
        displayName: 'Convert Response to Json',
        name: exports.CONVERT_TO_JSON,
        type: 'boolean',
        default: false,
        noDataExpression: true,
        description: 'Whether the SOAP response should be converted to JSON',
    },
];
